solutions/standard/main.tf

```diff
@@ -5,29 +5,33 @@
 module "resource_group" {
   source                       = "terraform-ibm-modules/resource-group/ibm"
   version                      = "1.1.5"
-  resource_group_name          = var.existing_resource_group == false ? var.resource_group_name : null
-  existing_resource_group_name = var.existing_resource_group == true ? var.resource_group_name : null
+  resource_group_name          = var.use_existing_resource_group == false ? var.resource_group_name : null
+  existing_resource_group_name = var.use_existing_resource_group == true ? var.resource_group_name : null
 }
 
 #######################################################################################################################
 # KMS Key
 #######################################################################################################################
 
 locals {
-  en_kms_key_id = var.existing_kms_root_key_id != null ? var.existing_kms_root_key_id : module.kms[0].keys[format("%s.%s", var.en_key_ring_name, var.en_key_name)].key_id
+  parsed_existing_kms_root_key_crn = var.existing_kms_root_key_crn != null ? split(":", var.existing_kms_root_key_crn) : []
+  existing_kms_root_key_id         = length(local.parsed_existing_kms_root_key_crn) > 0 ? local.parsed_existing_kms_root_key_crn[length(local.parsed_existing_kms_root_key_crn) - 1] : null
+  parsed_existing_kms_instance_crn = var.existing_kms_instance_crn != null ? split(":", var.existing_kms_instance_crn) : []
+  kms_region                       = length(local.parsed_existing_kms_instance_crn) > 0 ? local.parsed_existing_kms_instance_crn[5] : null
+  en_kms_key_id                    = local.existing_kms_root_key_id != null ? local.existing_kms_root_key_id : module.kms[0].keys[format("%s.%s", var.en_key_ring_name, var.en_key_name)].key_id
 }
 
 # KMS root key for Event Notifications
 module "kms" {
   providers = {
     ibm = ibm.kms
   }
-  count                       = var.existing_kms_root_key_id != null ? 0 : 1 # no need to create any KMS resources if passing an existing key
+  count                       = local.existing_kms_root_key_id != null ? 0 : 1 # no need to create any KMS resources if passing an existing key
```

Comment:
probably safer to use `var.existing_kms_root_key_id` here

Reply:
actually we removed `var.existing_kms_root_key_id` https://github.com/terraform-ibm-modules/terraform-ibm-event-notifications/pull/178/files#diff-e58a8d0a4f1a1a5eb0362e469eaf5ad7622f6dcb6f74df4683b3d8713c26ada1L85

Reply:
sorry I meant `existing_kms_root_key_crn`